scripts/covid19_india/cases_count_states_data/preprocess.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import csv
+import urllib.request
+
+# 1. All counts are at State/UT level
+# 2. Only cumulative counts are available at this point
+# 3. 'unassigned' cases are ignored
+# 5. Wrong state codes are ignored
+# 4. Sometimes reports are published more than once
+
+INDIAN_STATE_CODES = [
+    "IN-AP", "IN-AR", "IN-AS", "IN-BR", "IN-CT", "IN-GA", "IN-GJ", "IN-HR",
+    "IN-HP", "IN-JH", "IN-KA", "IN-KL", "IN-MP", "IN-MH", "IN-MN", "IN-ML",
+    "IN-MZ", "IN-NL", "IN-OR", "IN-PB", "IN-RJ", "IN-SK", "IN-TN", "IN-TG",
+    "IN-TR", "IN-UT", "IN-UP", "IN-WB", "IN-AN", "IN-CH", "IN-DN", "IN-DD",
```

Comment:
Just to confirm, with IN-DN and IN-DD merging, we still get separate stats?

I think IN-DD is missing in the KG, I'll take a TODO to add it.

Reply:
Good catch.

- They have published in all three formats.

- Initially they used to publish DD and DN separately. In between they started to publish them together as "Dadra and Nagar Haveli and Daman and Diu".  Volunteers have mapped this new entity an unofficial state code `DN_DD`.  ISOCode is TBA.

- In official vocab they are still listed separately - http://vocab.nic.in/cla.php?cat=mast&info=states

Reply:
Thanks for the explanation.

If we decide to get the merged data, we can temporarily attach DN_DD as the isoCode.

Reply:
I have added DN_DD as of now. I have also added a test case to check for the inclusion. Let me know if I need to do anything else on my side. I think we should include.

Reply:
Looks good.

Reply:
For  areas like IN-AR and IN-JK which are recognized as disputed territories, how do we ensure they are shown and labeled as PK-JK and CN-XJ as appropriate?

Reply:
Thanks for reminder on carefully dealing with disputed territories (a broader/general concern).

In the KG, these are represented by distinct entities:
- IN-JK (http://datacommons.org/browser/wikidataId/Q66278313)
- IN-AR (http://datacommons.org/browser/wikidataId/Q1162)
- CN-XJ (http://datacommons.org/browser/wikidataId/Q34800)
- PK-JK (http://datacommons.org/browser/wikidataId/Q200130)

This dataset is from Indian government, so it is reasonable that we attach them to the IN-xy entities.